rfcs/0114-nixos-code-of-conduct.md

```diff
@@ -0,0 +1,112 @@
+---
+feature: NixOS Code of Conduct
+start-date: 2021-11-03
+author: @jonringer
+co-authors:
+shepherd-team:
+shepherd-leader:
+related-issues: #98 #102
+---
+
+# Summary
+[summary]: #summary
+
+The NixOS Community needs some foundation of shared values which can be used
+to determine appropriate behavior when contributing and interacting with the community.
+We currently have a [community ethos](https://nixos.org/community/index.html), but
+this only states that we are an inclusive community, but doesn't define
+inappropriate behavior when interacting with the community. This RFC aims
+to establish the shared values, so they can be used to identify behavior
+which is disruptive to the community.
+
+This RFC does not intend to define moderation practices. This is intended to only lay
+a framework of shared values for the community.
+
+# Motivation
+[motivation]: #motivation
+
+The NixOS community lacks a clearly defined way in which someone's behavior may
+be disruptive. A well-defined Code of Conduct gives an explicit
+set of expectations for all contributors, which will also make it easier to identify
+when behavior becomes disruptive.
+
+# Detailed design
+[design]: #detailed-design
+
+These statutes are taken from [rust-lang's Code of Conduct](https://www.rust-lang.org/policies/code-of-conduct),
```

Comment:
I didn't want to comment this. I hoped I wouldn't need to, but here goes.

I think any CoC which lists specific "classes", or behaviours, or anything of the sort is bad. They will always end up being used for unfun rounds of "rules lawyering" from **any** side. This can already be observed in these comments. People are already LARPING "rules lawyering" about the CoC in this thread.

I have my own short list of CoCs I looked at that had better wording. I won't suggest any at this point in the hope that other individual look into other state of the art CoCs and consider then under that lens. I wouldn't want my personal choices to influence their investigation in the subject.

Reply:
I am not sure how this could be improved by being more generic or on a higher level because I am missing an example I could look at.

Maybe I am even totally misunderstanding what not listing any classes or groups of behavior in practice means.

Reply:
(Note: what I'm saying her is not specific to the chosen CoC, and may or may not apply. They are observed problems.)

Generally speaking, my opinion here is that if you list that A, B, and C are "protected classes", people will rules lawyer themselves reasons to say it's fine to discriminate against D and E since it's not listed.

Conversely, it may even be used as a way to bully "non-protected" "non-minorities", as the phrasing of such CoCs often makes it "rules lawyerable" to be okay to bully and harass anyone that's not listed.

It's hard to provide examples for this. I don't want to LARP-rules-lawyer bad scenarios, since it's not an easy thing for me to do. Additionally there's the risk of making up bad scenarios that will only be used to attack the ridiculousness of it, rather than the actual problem. Providing "real" examples is often unsavoury; I don't like digging up past drama in projects I've never been involved with, especially when I maybe don't know the full situation(?).

I just know that paradoxically, when you include things in a list, you invariably end up excluding.

Reply:
> Generally speaking, my opinion here is that if you list that A, B, and C are "protected classes", people will rules lawyer themselves reasons to say it's fine to discriminate against D and E since it's not listed.

That could probably be reduced by phrasing those section like ``for example A, B or C`` or like it is done a few lines below ``other similar characteristics``. I think it should be pretty easy to say that X is similar enough to B and be done with it.

If we dig up more such cases which don't really fit we can always add them later. I even think we should revisit this CoC at some point because we can't potentially cover all cases and everything beforehand.

Reply:
We already have people in these threads unhappy with these lists. It's clear enough that lists are not good. And here you are rules-lawyering about it in these replies already :/.

Reply:
> Generally speaking, my opinion here is that if you list that A, B, and C are "protected classes", people will rules lawyer themselves reasons to say it's fine to discriminate against D and E since it's not listed.
> 
> Conversely, it may even be used as a way to bully "non-protected" "non-minorities", as the phrasing of such CoCs often makes it "rules lawyerable" to be okay to bully and harass anyone that's not listed.

This is precisely the reason why I'm usually against any code of conducts. Everyone should be safe from harassment, but listing specific groups implicitly protects only them (Some common groups which are not listed in CoCs are political opinions and affiliations, and personal preferences). In addition, what is considered marginalized depends on region and culture. What might be a majority somewhere, could be a marginalized minority elsewhere.

I don't think we need to make a list of things, which are not allowed to be used as a basis of exclusion or harassment. Instead, I think no-one should feel excluded or harassed, and when exclusion or harassment inevitably occurs, mediating between the parties should be preferred to banhammering the guilty party.

Reply:
> Generally speaking, my opinion here is that if you list that A, B, and C are "protected classes", people will rules lawyer themselves reasons to say it's fine to discriminate against D and E since it's not listed.

Can you list the specific A Bs and Cs that are listed as “protected clases” by this RFC? I fail to see them.



> Everyone should be safe from harassment, but listing specific groups implicitly protects only them

This is just not true. It is a normal practice in law-like texts to list specific examples and then add a “or similar (in spirit)” catch-all at the end to signify that it’s not a closed-off list and up to the specific case (that’s also why you *need* judges to interpret the law and can’t just write a program as some nerds tend to believe).

As far as I can see, the linked rust CoC has the phrasing “[…] or other similar characteristic.” which is exactly such a clause.

Reply:
> LARPing 

Side note: what is LARPing? A quick google only revealed live action role play which seems to be unfitting.

Reply:
TL;DR what samueldr means is that someone was pretending to be a rules lawyer

Reply:
To be more precise, doing "rules lawyering", "role playing", on either imagined cases or theoretical cases.

Reply:
> This is just not true. It is a normal practice in law-like texts to list specific examples and then add a “or similar (in spirit)” catch-all at the end to signify that it’s not a closed-off list and up to the specific case (that’s also why you need judges to interpret the law and can’t just write a program as some nerds tend to believe).

Until they are invariably rules-lawyered either to work their way around consequences, or worse, *advocate harassment was okay*.

> Can you list the specific A Bs and Cs that are listed as “protected clases” by this RFC? I fail to see them.

Ugh... Assuming good faith from you, and not purposefully missing the point: 

> [...] level of experience, gender identity and expression, sexual orientation, disability, personal appearance, body size, race, ethnicity, age, religion (or lack thereof), socioeconomic status, nationality, or other similar characteristics.

And as much as I would like to assume we can trust the community to not do rules lawyering and making my point moot, first, it's not necessarily about *existing members*, which is something to remember, but really, it's about the fact that it seems we cannot assume basic conduct from anyone. If we were able to assume good basic conduct from everyone, the whole discussion here would be moot.

So under the assumption that we can't assume basic conduct, we can't have wording that can be abused. It is trivial for non-toxic persons to miss the point. And given your reaction, I assume you're probably non-toxic here!

* * *

The wording of a CoC cannot be done *in good faith* if the target audience is acting *in bad faith*. This might be a more distilled overview of my point of view.

Reply:
> The wording of a CoC cannot be done in good faith if the target audience is acting in bad faith. This might be a more distilled overview of my point of view.

Actually, maybe we should discuss who the target audience is. 

Personaly, I would consider a desirable outcome if (a) people switching between communities with significantly different norms can understand which mix we have or try to have here, and (b) so that a supermajority of contributors can come to an agreement that moderation is consistent with itself and with the written guidelines. Maybe also (c) the people taking some counter-actions, not exclusively moderators, can discuss propriety of those actions among themselves using a common vocabulary and common set of guidelines.

Indeed, (b) and (c) are open to rule-lawyering, but not as much to malicious rule-lawyering.